changelogs/client_server/newsfragments/1909.clarification

```diff
@@ -0,0 +1 @@
+The `UserIdentifier` object in `POST /_matrix/client/v3/login` can contain additional properties.
```

Comment:
I believe that a better solution would be to change the definition of `UserIdentifier` rather than encourage the use of `true`.

It would probably be good to always have a `description` for `additionalProperties`.

In this particular case, we could also define that the values of the additional properties are always strings.

Reply:
I thought about that but without a technical way to prohibit the use of `additionalProperties: true` I was worried that it might easily creep in again elsewhere in the spec in the future?

Reply:
We could add a bright red error banner in the HTML if `additionalProperties` is set to something other than the desired type? That would at least alert content writers to their mistake.

Hugo doesn't look like it has any `assert` functionality to fail the build otherwise.

Reply:
Hugo has one, it's [`errorf`](https://gohugo.io/functions/fmt/errorf/).

Reply:
@Johennes can you modify this PR to invoke `errorf` in this case?

Reply:
Apologies for the delay I was away for a few days. The `errorf` function sounds brilliant. I didn't know this existed. I have changed things around to enforce `additionalProperties: {}` instead.